gtsam/linear/GaussianFactorGraph.h

```diff
@@ -180,20 +180,84 @@ namespace gtsam {
     ///@name Linear Algebra
     ///@{
 
+    /// Sparse matrix representation as vector of tuples.
+    typedef std::vector<boost::tuple<size_t, size_t, double>>
+        SparseMatrixBoostTriplets;
+    /// Sparse matrix representation as vector of slightly more efficient
+    /// tuples.
+    typedef std::vector<std::tuple<int, int, double>> SparseMatrixGtsamTriplets;
+
     /**
-     * Return vector of i, j, and s to generate an m-by-n sparse Jacobian matrix,
-     * where i(k) and j(k) are the base 0 row and column indices, s(k) a double.
+     * Return vector of i, j, and s to generate an m-by-n sparse augmented
+     * Jacobian matrix, where i(k) and j(k) are the base 0 row and column
+     * indices, s(k) a double.
      * The standard deviations are baked into A and b
+     * @param ordering the column ordering
+     * @param[out] nrows The number of rows in the Jacobian
+     * @param[out] ncols The number of columns in the Jacobian
+     * @return the sparse matrix in one of the 4 forms above
      */
-    std::vector<boost::tuple<size_t, size_t, double> > sparseJacobian() const;
+    SparseMatrixBoostTriplets sparseJacobian(const Ordering& ordering,
+                                             size_t& nrows,
+                                             size_t& ncols) const;
+
+    /** Returns a sparse augmented Jacobian without outputting its dimensions */
+    SparseMatrixBoostTriplets sparseJacobian(
+        const Ordering& ordering) const;
+
+    /** Returns a sparse augmented Jacobian with default Ordering */
+    SparseMatrixBoostTriplets sparseJacobian(size_t& nrows,
+                                             size_t& ncols) const;
+
+    /** Returns a sparse augmented Jacobian without with default ordering and
+     * outputting its dimensions */
+    SparseMatrixBoostTriplets sparseJacobian() const;
```

Comment:
Why do we all of a sudden have that many versions? Which ones do we need *now* ? Which are built "for the future"?

Reply:
Right now, only need to add the one with all the arguments:
`SparseMatrixBoostTriplets sparseJacobian(const Ordering& ordering,
                                             size_t& nrows,
                                             size_t& ncols) const;`

I'll remove the others.

I should note though that I'm not sure how people are "supposed" to use the version without the `nrows` and `ncols` arguments since otherwise you cannot tell how large the real matrix is (there may be zero columns/zero rows at the end that you would never know about).  But I guess we'll leave that up to users to figure out.

Reply:
done